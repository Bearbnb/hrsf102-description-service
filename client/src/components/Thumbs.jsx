import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../../styles/HomeHighlight.css';


const Thumbs = ({ handleClick }) => (
  <div className={styles.helpful}>

    <button
      className={styles.button}
      type="button"
      onClick={() => handleClick()}
    >Helpful
      <i className="far fa-thumbs-up" />
    </button>

    <span className={styles.interpunct}>Â·</span>

    <button
      className={styles.button}
      type="button"
      onClick={() => handleClick()}
    >Not Helpful
    </button>

  </div>
);

Thumbs.propTypes = {
  handleClick: PropTypes.func,
};

Thumbs.defaultProps = {
  handleClick: () => {},
};

export default Thumbs;
