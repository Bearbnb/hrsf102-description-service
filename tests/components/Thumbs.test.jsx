import { shallow } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Thumbs from '../../client/src/components/Thumbs';

describe('<Thumbs />', () => {
  it('should render properly', () => {
    const wrapper = shallow(<Thumbs />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
