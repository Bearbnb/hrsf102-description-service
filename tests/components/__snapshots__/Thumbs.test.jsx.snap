// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Thumbs /> should render properly 1`] = `
<div
  className="helpful"
>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Helpful
    <i
      className="i far fa-thumbs-up"
    />
  </button>
  <span
    className="interpunct"
  >
    Â·
  </span>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    Not Helpful
  </button>
</div>
`;
